magicctapipe/scripts/lst1_magic/setting_up_config_and_dir.py

```diff
@@ -0,0 +1,465 @@
+    
+    ###########################################
+    ##################### MC
+    ###########################################
+        
+    if not os.path.exists(target_dir):
+        os.mkdir(target_dir)
+        os.mkdir(target_dir+"/DL1")
+        os.mkdir(target_dir+"/DL1/Observations")
+        os.mkdir(target_dir+"/DL1/MC")
+        os.mkdir(target_dir+"/DL1/MC/gammas")
+        os.mkdir(target_dir+"/DL1/MC/gammadiffuse")
+        os.mkdir(target_dir+"/DL1/MC/electrons")
+        os.mkdir(target_dir+"/DL1/MC/protons")
+        os.mkdir(target_dir+"/DL1/MC/helium")
+    else:
+        overwrite = input("MC directory for "+target_dir.split("/")[-1]+" already exists. Would you like to overwrite it? [only 'y' or 'n']: ")
+        if overwrite == "y":
+            os.system("rm -r "+target_dir)
+            os.mkdir(target_dir)
+            os.mkdir(target_dir+"/DL1")
+            os.mkdir(target_dir+"/DL1/Observations")
+            os.mkdir(target_dir+"/DL1/MC")
+            os.mkdir(target_dir+"/DL1/MC/gammas")
+            os.mkdir(target_dir+"/DL1/MC/gammadiffuse")
+            os.mkdir(target_dir+"/DL1/MC/electrons")
+            os.mkdir(target_dir+"/DL1/MC/protons")
+            os.mkdir(target_dir+"/DL1/MC/helium")
+        else:
+            print("Directory not modified.")
+    
+    
+    
+    ###########################################
+    ##################### MAGIC
+    ###########################################
+    
+    if telescope_ids[-1] > 0:    
+        if not os.path.exists(target_dir+"/DL1/Observations/M2"):
+            os.mkdir(target_dir+"/DL1/Observations/M2")
+            for i in MAGIC_runs:
+                if not os.path.exists(target_dir+"/DL1/Observations/M2/"+i[0]):
+                    os.mkdir(target_dir+"/DL1/Observations/M2/"+i[0])
+                    os.mkdir(target_dir+"/DL1/Observations/M2/"+i[0]+"/"+i[1])
+                else:
+                    os.mkdir(target_dir+"/DL1/Observations/M2/"+i[0]+"/"+i[1])
+    
+    if telescope_ids[-2] > 0:
+        if not os.path.exists(target_dir+"/DL1/Observations/M1"):
+            os.mkdir(target_dir+"/DL1/Observations/M1")
+            for i in MAGIC_runs:
+                if not os.path.exists(target_dir+"/DL1/Observations/M1/"+i[0]):
+                    os.mkdir(target_dir+"/DL1/Observations/M1/"+i[0])
+                    os.mkdir(target_dir+"/DL1/Observations/M1/"+i[0]+"/"+i[1])
+                else:
+                    os.mkdir(target_dir+"/DL1/Observations/M1/"+i[0]+"/"+i[1])
+    
+
+
+
+
+def main():
+
+    """ Here we read the config_general.yaml file and call the functions to generate the necessary directories, bash scripts and launching the jobs."""
+    
+    parser = argparse.ArgumentParser()
+    
+    #Here we are simply collecting the parameters from the command line, as input file, output directory, and configuration file
+    parser.add_argument(
+        "--partial-analysis",
+        "-p",
+        dest="partial_analysis",
```

Comment:
I would add here a `choices` list as well. Moreover, `--partial-analysis` as name is a bit reductive right? I mean, if you choose `onlyMAGIC` it will perform all the analysis only with MAGIC right? Maybe `analysis_type` or similar. Or, even get rid of this argument and have the choice available from the config file. You have a better view of the big picture, so I let you decide which one is more suitable.